architecture: update membership and roles description (#152)
pingcap/community: architecture/README.md

        * [Privileges and responsibilities](#responsibilities-and-privileges-2)
* [TiDB User Group](#tidb-user-group)
* [Organizer Committee](#organizer-committee)

<!-- vim-markdown-toc -->

## PMC

PMC (Project Management Committee) functions as the core management team that oversees the TiDB community. See [Governance](../GOVERNANCE.md) for more details about PMC and [PMC List](./pmc-list.md) for a list of incumbent PMC members.

## TiDB Developer Group

As the cornerstone of community development, the TiDB Developer Group consists of these roles: Maintainer, Committer, Reviewer, Active Contributor, and Contributor. Each role takes corresponding responsibilities in the community. Collectively they play an important role in the robust development of TiDB.

### Contributor

Anyone who has one PR merged for any project in the [PingCAP Organization](https://github.com/pingcap) is a Contributor. New Contributors should be welcomed to the community by existing members, helped with PR workflow, and directed to relevant documentation and communication channels. A contributor becomes a member of the SIG of the contribution automatically.

#### How to become a Contributor

> - merged at least 1 PR for any project in the [PingCAP Organization](https://github.com/pingcap)

Reviewer: ```suggestion
- At least 1 PR being merged for any project in the [PingCAP Organization](https://github.com/pingcap)
```

Author: "merged" here is a consistent usage with other requirements. Here the full sentence is intended as "(You must) have...

You are also encouraged to participate in the projects in the following ways:

- Actively answer technical questions asked by community users in GitHub issues.
- Help test the projects
- Help review the pull requests (PRs) submitted by others
- Help improve technical documents
- Submit valuable issues
- Report or fix known and unknown bugs
- Write articles about source code analysis and usage cases for a project.

#### Contributor List

See [TiDB Contributors](./contributor-list.md)

### Active Contributor

> Active Contributors are continuously active contributors in the community. They can have issues and PRs assigned to them, participate in SIGs through GitHub teams, and pre-submit tests are automatically run for their PRs. See `member-list.md` under the specific SIG for a list of Active Contributors.

Reviewer: > Active Contributors are continuously active contributors in the community
How about we change it into "Active Contributors are those who continuously and actively make contributions in the community"

Author: It means the same, only longer.

> Active Contributors are continuously active contributors in the community. They can have issues and PRs assigned to them, participate in SIGs through GitHub teams, and pre-submit tests are automatically run for their PRs. See `member-list.md` under the specific SIG for a list of Active Contributors.

Author: ```suggestion
Active Contributors are contributors who make continuous contributions in the community. They can have issues and PRs assigned to them, participate in SIGs through GitHub teams, and pre-submit tests are automatically run for their PRs. See `member-list.md` under the specific SIG for a list of Active Contributors.
```

#### How to become a Maintainer

- Demonstrated a deep and comprehensive understanding of TiDB's technical goals and directions
- Must be Committer for at least 2 SIGs
- Actively involved in organizing technical seminars and proposing or reviewing major technical designs
- Can handle complex problems in the code implementation process
- Nominated by and obtained supermajority votes from the PMC

#### Privileges and responsibilities

- Develop and approve technical design and decisions for TIDB projects
- Set the technical directions and priorities for SIG projects
- Participate in product releases and roadmap development of the project
- Continuously contribute to the project by mentoring Reviewers and Committers
- Ensure the overall quality of the TiDB project

## TiDB User Group

The TiDB User Group (TUG) is an independent user community initiated by TiDB users. Users in the TUG community can learn cutting-edge technical knowledge, publish technical insights, get to know friends in the circle, and build TiDB projects together.

### Leader

> The Leader of the  TiDB User Group (TUG) who makes overall plans, tracks progress to strengthen the joint force and impact of the TUG.

Reviewer: ```suggestion
The Leader of the  TiDB User Group (TUG) is who makes overall plans, tracks progress to strengthen the joint force and impact of the TUG.
```

Author: No "is" needed. This is not a full sentence. It's quite normal to explain some concepts using this structure.

Reviewer: I see~

### Co-Leader

Co-Leaders share TUG responsibilities such as development planning, event operation, content creation, and community Q&A.

### Ambassador

> Community activists who recognize TiDB and are passionate about sharing and promoting TiDB and TUG on bloggers, community websites, and meetups.

Reviewer: ```suggestion
TUG Ambassadors are community activists who recognize TiDB and are passionate about sharing and promoting TiDB and TUG on bloggers, community websites, and meetups.
```

## TiDB Developer Group

As the cornerstone of community development, the TiDB Developer Group consists of these roles: Maintainer, Committer, Reviewer, Active Contributor, and Contributor. Each role takes corresponding responsibilities in the community. Collectively they play an important role in the robust development of TiDB.

### Contributor

New contributors should be welcomed to the community by existing members, helped with PR workflow, and directed to relevant documentation and communication channels.

#### How to become a Contributor

- merged at least 1 PR for any project in the [PingCAP Organization](https://github.com/pingcap)

You are also encouraged to participate in the projects in the following ways:

- Actively answer technical questions asked by community users in GitHub issues.
- Help test the projects
- Help review the pull requests (PRs) submitted by others
- Help improve technical documents
- Submit valuable issues
- Report or fix known and unknown bugs
- Write articles about source code analysis and usage cases for the - projects.

> #### Responsibilities and privileges

Author: ```suggestion
```

> - Become a member in the SIG

Reviewer 2: I'm not sure whether it's proper to be a responsibility of contributor.

Author: Here it's a privilege

- Get the profile listed on the membership page of the TiDB community

#### Contributor List

See [TiDB Contributors](./contributor-list.md)

### Active Contributor

> Active Contributors are continuously active contributors in the community. They can have issues and PRs assigned to them, participate in SIGs through GitHub teams, and pre-submit tests are automatically run for their PRs. See `member-list.md` under the specific SIG for a list of Active Contributors.

Reviewer 2: > They can have issues and PRs assigned to them

I think everyone can be assigned to an issue, no matter whether they're contributors.

Author: Basically I don't think Contributors are not established or committed enough to be assigned with tasks. They are contributing out of their willingness or interest. You can suggest or recommend, but not assign  a task to them (assigning means obligations or responsibilities, which they do not have yet )

  - Merged 20+ PRs (may vary among SIGs)
  - Handled 1+ more issue of Medium difficulty or higher
- Sponsored by at 2 Committers of the SIG

#### Responsibilities and Privileges

- Ensure the correctness of the code submitted to the SIG
- Respond promptly to PRs
- Participate in SIG meetings regularly
- Approve code reviews with LGTM (Looks Good To Me)
- Get the profile listed on the membership page of the corresponding SIG

### Committer

Committers come from Reviewers, and are recoganized by Maintainers and SIG Tech Leads for their outstanding contributions to the specific SIG.

Committer is scoped to the specific SIG where the promotion applies. See `member-list.md` under the specific SIG for a list of Committers of the SIG.

#### How to become a Committer

A Committer must have accomplished one or more of the following items:

- Demonstrated a good sense of responsibility in PR reviews within the SIG. The review contributions must meet the amount or the difficulty criteria as defined in the charter of the specific SIG.
- Demonstrated deep understanding of the TiDB components specific to the SIG by contributing significantly as:

>     - Finished 2+ tasks of Medium difficulty or higher

Author: ```suggestion
    - Finished 2 or more tasks of Medium difficulty
```

>     - Fixed 1+ task of Hard difficulty

Reviewer 3: `1+` means `one or larger than one` or `larger than one`?

Author: Good point. I updated it to "1 or more"

> - Nominated by the Tech Leads of the SIG and gained supermajority (2/3) votes from Maintainers

Author: @zz-jason Please look at this part again, and make sure the rules or requirements are suitable. Also - we need to add SIG specific charters

Reviewer 2: OK, I'll take a look this evening.

Reviewer 2: looks good to me.

#### Responsibilities and privileges

- Control overall code quality of projects under the SIG
- Guide Contributors and Reviewers to contribute to the SIG continuously
- Participate in Committer meetings regularly and have knowledge priority regarding project development
- Participate in design discussions
- Get the profile listed on the membership page of the corresponding SIG

### Maintainer

> Maintainers are the planners and designers of the TiDB project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the sustained development and overall health of the TiDB project. See [TiDB Maintainers](./maintainer-list.md) for a list of TiDB Maintainers.

Reviewer 2: the url for maintain list is broken,

Author: I think it has to be merged to be effective because the maintainer list file is in master, not this branch.

> Maintainers are the planners and designers of the TiDB project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the sustained development and overall health of the TiDB project. See [TiDB Maintainers](./maintainer-list.md) for a list of TiDB Maintainers.

Author: ```suggestion
Maintainers are the planners and designers of the TiDB project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the sustained development and overall health of the TiDB project. See [TiDB Maintainers](/maintainer-list.md) for a list of TiDB Maintainers.
```

## PMC

> PMC (Project Management Committee) functions as the core management team that oversees the TiKV community. The PMC has additional responsibilities over and above those of a committer. These responsibilities ensure the smooth running of the project. PMC members are expected to review code contributions, participate in strategic planning, approve changes to the governance model and manage the copyrights within the project outputs. See [Governance](./GOVERNANCE.md) for more details about PMC.

Reviewer 2: It should be `See [Governance](../GOVERNANCE.md) for more details about PMC.`

You can click `view file` to preview this page to check whether is a dead link. We need a dead link detecter. 🤔

<img width="252" alt="Screen Shot 2019-12-28 at 8 21 00 PM" src="https://user-images.githubusercontent.com/5268763/71543587-a3106e00-29af-11ea-821c-183acff6b0dc.png">

## TiDB Developer Group

As the cornerstone of community development, the TiDB Developer Group consists of these roles: Maintainer, Committer, Reviewer, Active Contributor, and Contributor. Each role takes corresponding responsibilities in the community. Collectively they play an important role in the robust development of TiDB.

### Contributor

New contributors should be welcomed to the community by existing members, helped with PR workflow, and directed to relevant documentation and communication channels.

#### How to become a Contributor

> - Merged at least 1 PR for any SIG

Reviewer 2: how about:
```
merged at least 1 PR for any project in the [PingCAP Organization](https://github.com/pingcap)
```

See `member-list.md` under the specific SIG for a list of Reviewers of the SIG.

#### How to become a Reviewer

- Familiar with the codebase
- Demonstrated a recoganized level of community engagement in the SIG development, via contributions that have met one or more of the minimal criteria as defined below：
  - Merged 20+ PRs (may vary among SIGs)
  - Handled 1+ more issue of Medium difficulty or higher
- Sponsored by at 2 Committers of the SIG

#### Responsibilities and Privileges

- Ensure the correctness of the code submitted to the SIG
- Respond promptly to PRs
- Participate in SIG meetings regularly
- Approve code reviews with LGTM (Looks Good To Me)
- Get the profile listed on the membership page of the corresponding SIG

### Committer

Committers come from Reviewers. Committer is scoped to the specific SIG where the promotion applies. Committers are recoganized by Maintainers or SIG Tech Leads for their outstanding contributions to the specific SIG.

Committer status is scoped to the part of the codebase under the SIG. See `member-list.md` under the specific SIG for a list of Committers of the SIG.

#### How to become a Committer

> A new Committer must be nominated by any existing Maintainer or PMC member, and

Reviewer 4: Committer should be sponsored by 2 maintainers, but not necessarily voted by PMC

* You have implemented a complete feature.
* You have fixed a vital bug.

#### Responsibilities and privileges

- Control overall code quality of projects under the SIG

> - Guide Contributors and Reviewers to contribute to TiKV continuously

Reviewer 4: ```suggestion
- Guide Contributors and Reviewers to contribute to TiDB continuously
```

- Participate in Committer meetings regularly and have knowledge priority regarding project development
- Participate in design discussions
- Get the profile listed on the membership page of the corresponding SIG

### Maintainer

Maintainers are the planners and designers of the TiDB project, with the authority to merge branches into the master. Maintainers come from Committers. They must demonstrate both good judgment and responsibility towards the sustained development and overall health of the TiDB project. See [TiDB Maintainers](./maintainer-list.md) for a list of TiDB Maintainers

#### How to become a Maintainer

> - Demonstrated a deep and comprehensive understanding of TiKV's technical goals and directions

Reviewer 4: ```suggestion
- Demonstrated a deep and comprehensive understanding of TiDB's technical goals and directions
```

- Must be Committer for at least 2 SIGs
- Actively involved in organizing technical seminars and proposing or reviewing major technical designs
- Can handle complex problems in the code implementation process
- Nominated by and obtained supermajority votes from the PMC

#### Responsibilities and Privileges

> - Develop and approve technical design and decisions for TiKV projects

Reviewer 4: ```suggestion
- Develop and approve technical design and decisions for TiDB projects
```
